core/modules/comment/comment.install

```diff
@@ -470,6 +470,26 @@ function comment_update_1007() {
   }
 }
 
+/**
+ * New setting for comment titles. Replacing comment_title_field
+ * with comment_subject_options.
+ */
+function comment_update_1008() {
+  foreach (config_get_names_with_prefix('node.type.') as $config_name) {
+    $config = config($config_name);
+    $comment_title = $config->get('settings.comment_subject_field');
+    if ($comment_title == COMMENT_TITLE_AUTO) {
+      // auto generated titles
+      $config->set('settings.comment_title_options', COMMENT_TITLE_AUTO);
+    } else {
+      // custom titles
+      $config->set('settings.comment_title_options', COMMENT_TITLE_CUSTOM);
+    }
+    $config->clear('settings.comment_subject_field');
```

Comment:
@BWPanda - I am not sure what you are suggesting here - or is it possible I already implemented your suggestion?

Reply:
It's a bit complicated... `hook_uninstall()` removes variables from `$node_type->settings`. But since the old `subject` variable was deleted from `hook_uninstall()`, it'll need to be removed from `$node_type->settings` somewhere else then. `comment_update_1008()` seems appropriate since this is where you're replacing the old `subject` variable with the new `title` one. So here you need to add similar code to what's in hook_uninstall()`, but just for the `subject` variable. Hope that makes sense.

Reply:
I think it does. Processing now...

Reply:
@BWPanda - I think got it. BUT, I'm stretching here and could easily be missing things. I would appreciate another look.

Reply:
Looks good to me!